docs/en/Distributed-Lock.md

```diff
@@ -0,0 +1,72 @@
+# Distributed Lock
+Distributed locks are very useful to manage many processes that request the same object. 
+They are used to handle conflict between these requests. Once obtaining anyone it, the others need to wait till give up free.
+
+# Providers
+`MedallionAbpDistributedLock`
+Distributed locks system provides an abstraction that can be implemented by any vendor/provider.
+ABP depends on [DistributedLock.Core](https://www.nuget.org/packages/DistributedLock.Core) library which provides a distributed locking system for concurrency control in a distributed environment. There are [many distributed lock providers](https://github.com/madelson/DistributedLock#implementations) including Redis, SqlServer and ZooKeeper. You may use the one you want. Here, I will show the Redis provider.
+
+This provider contains a method named TryAcquireAsync and this method returns null if the lock could not be handled.
+> Name is a mandatory field. It keeps the locked provider name.
+> Timeout is set as default. If it fells deadlock and doesn't work properly you can use define the time to kill it.
+>CancellationToken is set as default. It enables cooperative cancellation between threads, thread pool work items, or Task objects
+
+
+Also, you should add [DistributedLock.Redis](https://www.nuget.org/packages/DistributedLock.Redis) NuGet package to your project, then add the following code into the ConfigureService method of your ABP module class.
+
+````csharp
+using Medallion.Threading;
+using Medallion.Threading.Redis;
+namespace AbpDemo
+{
+	public class MyModule : AbpModule
+	{
+		public override void ConfigureServices(ServiceConfigurationContext context)
+		{
+			//the other configurations
+			var configuration = context.Services.GetConfiguration();
+			
+			context.Services.AddSingleton<IDistributedLockProvider>(sp =>
+			{
+				var connection = ConnectionMultiplexer.Connect(configuration["Redis:Configuration"]);
+				return new RedisDistributedSynchronizationProvider(connection.GetDatabase());
+			});
+		}
+	}
+}
+````
+
+Also, you should add your Redis configuration in appsetting.json
+````json
+"Redis": {
+    "Configuration": "127.0.0.1"
+}
+````
+
+> To use in APB, you should download the below NuGet package.
```

Comment:
I don't think this and the below lines for installing the package are necessary. Maybe we can create an **Installation** section as in [Redis-Cache documentation](https://docs.abp.io/en/abp/latest/Redis-Cache#installation) and add the following ABP CLI command to the section as an installation command.

```bash
abp add-package Volo.Abp.DistributedLocking
```